lib/html.php

```diff
@@ -51,3 +51,95 @@ function defaultLinkTo($content, $server){
+	while(strpos($string, $start) !== false) {
+		$section_to_remove = substr($string, strpos($string, $start));
+		$section_to_remove = substr($section_to_remove, 0, strpos($section_to_remove, $end) + strlen($end));
+		$string = str_replace($section_to_remove, '', $string);
+	}
+	return $string;
+}
+
+/**
+ * Remove HTML sections containing one or more sections using the same HTML tag
+ * @param $string input string, e.g. 'foo<div class="ads"><div>ads</div>ads</div>bar'
+ * @param $tag_name name of the HTML tag, e.g. 'div'
+ * @param $tag_start start of the HTML tag to remove, e.g. '<div class="ads">'
+ * @return cleaned string, e.g. 'foobar'
+ */
+function stripRecursiveHTMLSection($string, $tag_name, $tag_start){
+	$open_tag = '<' . $tag_name;
+	$close_tag = '</' . $tag_name . '>';
+	$close_tag_length = strlen($close_tag);
+	if(strpos($tag_start, $open_tag) === 0) {
+		while(strpos($string, $tag_start) !== false) {
+			$max_recursion = 100;
+			$section_to_remove = null;
+			$section_start = strpos($string, $tag_start);
+			$search_offset = $section_start;
+			do {
+				$max_recursion--;
+				$section_end = strpos($string, $close_tag, $search_offset);
+				$search_offset = $section_end + $close_tag_length;
+				$section_to_remove = substr($string, $section_start, $section_end - $section_start + $close_tag_length);
+				$open_tag_count = substr_count($section_to_remove, $open_tag);
+				$close_tag_count = substr_count($section_to_remove, $close_tag);
+			} while ($open_tag_count > $close_tag_count && $max_recursion > 0);
+			$string = str_replace($section_to_remove, '', $string);
+		}
+	}
+	return $string;
+}
+
+/**
+ * Convert Markdown tags into HTML tags. Only a subset of the Markdown syntax is implemented.
+ * @param $string input string in Markdown format
+ * @return output string in HTML format
+ */
+function markdownToHtml($string) {
+	$string = preg_replace('/\!\[([^\]]+)\]\(([^\) ]+)(?: [^\)]+)?\)/', '<img src="$2" alt="$1" />', $string);
+	$string = preg_replace('/\[([^\]]+)\]\(([^\)]+)\)/', '<a href="$2">$1</a>', $string);
+	$string = preg_replace('/\*\*(.*)\*\*/U', '<b>$1</b>', $string);
+	$string = preg_replace('/\*(.*)\*/U', '<i>$1</i>', $string);
+	$string = preg_replace('/__(.*)__/U', '<b>$1</b>', $string);
+	$string = preg_replace('/_(.*)_/U', '<i>$1</i>', $string);
+	$string = preg_replace('/[-]{6,99}/', '<hr />', $string);
+	$string = str_replace('&#10;', '<br />', $string);
+	$string = preg_replace('/([^"])(https?:\/\/[^ "<]+)([^"])/', '$1<a href="$2">$2</a>$3', $string.' ');
+	$string = preg_replace('/([^"\/])(www\.[^ "<]+)([^"])/', '$1<a href="http://$2">$2</a>$3', $string.' ');
+
+	$count = 1;
+	while($count > 0) {
+		$string = preg_replace('/ (src|href)="([^"]+)<i>([^"]+)"/U', ' $1="$2_$3"', $string, -1, $count);
+	}
+
+	$count = 1;
+	while($count > 0) {
+		$string = preg_replace('/ (src|href)="([^"]+)<\/i>([^"]+)"/U', ' $1="$2_$3"', $string, -1, $count);
+	}
+
+	return '<div>'.trim($string).'</div>';
```

Comment:
Thanks for the details. I agree that adding a dependency for Markdown support is not worth it right now. Having this function available, however, is very nice :+1: 

It would be great if you could add the links to the source code as well, so we know how it works in a year or two :smile: 

This function might become handy for other bridges as well (especially for GitHub and others that rely on Markdown), although I suppose some sites don't rely on JS that much.

Reply:
OK, added that in a1dd50a89b6685161b68d4afaa4cdcda8527e437 :slightly_smiling_face: